diff --git a/lib/CppMicroServices-development/CMakeLists.txt b/lib/CppMicroServices-development/CMakeLists.txt
index 73e8d34..ade346c 100644
--- a/lib/CppMicroServices-development/CMakeLists.txt
+++ b/lib/CppMicroServices-development/CMakeLists.txt
@@ -506,9 +506,9 @@ endif()
 add_subdirectory(third_party/absl)
 #-----------------------------------------------------------------------------

-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${US_CXX_FLAGS}")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${US_CXX_FLAGS} -fPIC -Wno-error=deprecated-declarations")
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${US_CXX_FLAGS_RELEASE}")
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${US_C_FLAGS}")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${US_C_FLAGS} -fPIC")
 set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} ${US_C_FLAGS_RELEASE}")

 #-----------------------------------------------------------------------------
diff --git a/lib/CppMicroServices-development/compendium/DeclarativeServices/src/metadata/ReferenceMetadata.cpp b/lib/CppMicroServices-development/compendium/DeclarativeServices/src/metadata/ReferenceMetadata.cpp
index 7a2ac74..0c8fbe9 100644
--- a/lib/CppMicroServices-development/compendium/DeclarativeServices/src/metadata/ReferenceMetadata.cpp
+++ b/lib/CppMicroServices-development/compendium/DeclarativeServices/src/metadata/ReferenceMetadata.cpp
@@ -23,6 +23,8 @@
 #include "ReferenceMetadata.hpp"
 #include "Util.hpp"

+#include <limits>
+
 namespace cppmicroservices {
 namespace scrimpl {
 namespace metadata {
diff --git a/lib/CppMicroServices-development/compendium/tools/SCRCodeGen/ComponentInfo.cpp b/lib/CppMicroServices-development/compendium/tools/SCRCodeGen/ComponentInfo.cpp
index 21c762d..9d68867 100644
--- a/lib/CppMicroServices-development/compendium/tools/SCRCodeGen/ComponentInfo.cpp
+++ b/lib/CppMicroServices-development/compendium/tools/SCRCodeGen/ComponentInfo.cpp
@@ -20,6 +20,7 @@

   =============================================================================*/

+#include <iterator>
 #include <regex>
 #include <sstream>

